packages/nimble-components/src/table-column/base/tests/table-column.fixtures.ts

```diff
@@ -11,6 +11,11 @@ import type {
 import { ColumnValidator } from '../models/column-validator';
 
 export const tableColumnEmptyCellViewTag = 'nimble-test-table-column-empty-cell-view';
+declare global {
+    interface HTMLElementTagNameMap {
+        [tableColumnEmptyCellViewTag]: TableCellView;
```

Comment:
Could you note why the syntax here is different, with the tag surrounded by `[]`?

Reply:
FAST has two patterns for component registration, `compose()` and the `@customElement` decorator. I'd like to use just the decorator but it doesn't support overriding foundation configuration like compose does. So we could just use compose (which is needlessly verbose for the test fixtures) or mix both.

Another difference is that the decorator takes the full tag name while compose takes the prefix (nimble-) and base tag separately.

So to remove the constants in a compose based file I'd like to actually do (maybe also importing the prefix):

```ts
const baseName = 'anchor-button';
export const anchorButtonTag = `nimble-${baseName}`;
declare global {
    interface HTMLElementTagNameMap {
        [anchorButtonTag]: AnchorButton;
    }
}

const nimbleAnchorButton = AnchorButton.compose<AnchorOptions>({
    baseName,
});

DesignSystem.getOrCreate().withPrefix('nimble').register(nimbleAnchorButton());
```

And in modern version of typescript that template expression used for a type string is allowed ([see playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBARgQwgUwHIILbJgXhgcgTGAAsQAnAWjgFcopx8BuAKGQA8AHC2USWIqQoAhOgzAAVBAHNcMAAZgAlhjgAbZJQAkAb0Qp0WAL7zWAE2TA1CctmlqQiNTB0sY7mErBRk5AGYIwNgAEhIAsgAyAKIaWN5S0obIYQicLm4emQDagmTkovTgCQC6AFwwAILEeQXirJlGLI1WSBCV1SJi4C6NQA)) and calculated at compile time. I tried it in [this branch](https://github.com/ni/nimble/compare/main...strict-type-names#diff-276a46796655c4f0352e76dc16ce29522889d60dd46a12ec73073be024be4218R13) but it failed in the current version of typescript.

So ideally in the compose based files we would do the snippet above and in decorator based files (which we already can do today since no expression needs to be evaluated):
```ts
export const anchorTag= 'nimble-anchor';
declare global {
    interface HTMLElementTagNameMap {
        [anchroTag]: Anchor;
    }
}
@customElement({
    name: anchorTag
})
class Anchor {}
```

Soo we can use the decorator pattern today as you noticed and we can switch to the slightly better compose pattern if we cared in the future. This PR shows the quickest change to get the type improvements today but doesn't reduce the string duplication in component definition yet (we could reduce it slightly but doesn't seem worth it, would rather get it to the single sourced state after TS upgrade).